agent/engine/engine_integ_test.go

```diff
@@ -284,3 +283,32 @@ func TestStartStopWithCredentials(t *testing.T) {
 	_, ok := credentialsManager.GetTaskCredentials(credentialsIDIntegTest)
 	assert.False(t, ok, "Credentials not removed from credentials manager for stopped task")
 }
+
+func TestContainerHealthCheck(t *testing.T) {
+	taskEngine, done, _ := setupWithDefaultConfig(t)
+	defer done()
+
+	stateChangeEvents := taskEngine.StateChangeEvents()
+
+	taskArn := "testTaskWithHealthCheck"
+	testTask := createTestHealthCheckTask(taskArn)
+
+	go taskEngine.AddTask(testTask)
+
+	event := <-stateChangeEvents
```

Comment:
you can use the `verify[Container/Task][Running|Stopped]` here and in the subsequent block.